src/client/mapblock_mesh.cpp

```diff
@@ -1303,19 +1306,23 @@ MapBlockMesh::MapBlockMesh(MeshMakeData *data, v3s16 camera_offset):
 				// Dummy sunlight to handle non-sunlit areas
 				video::SColorf sunlight;
 				get_sunlight_color(&sunlight, 0);
-				u32 vertex_count = p.vertices.size();
+
+				std::map<u32, video::SColor> colors;
```

Comment:
How about
```C++
auto &colors = m_daynight_diffs[std::pair<u8, u32>(layer, i)];
colors.clear();
// loop
```
.. or is it necessary to keep existing map contents?

Reply:
I want to avoid creating a map when there is no contents. this could make a difference, see here:
https://github.com/minetest/minetest/blob/87472150bcb83e9cbad2f567ac536de0456ceb70/src/client/mapblock_mesh.cpp#L1384-L1387